packages/libs/error-stack/src/frame/kind.rs

```diff
@@ -1,15 +1,39 @@
+use core::fmt::{Debug, Display};
+
+use crate::Context;
+
 /// Classification of the contents of a [`Frame`], determined by how it was created.
 ///
 /// [`Frame`]: crate::Frame
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
-pub enum FrameKind {
+pub enum FrameKind<'f> {
     /// Frame was created through [`Report::new()`] or [`change_context()`].
     ///
     /// [`Report::new()`]: crate::Report::new
     /// [`change_context()`]: crate::Report::change_context
-    Context,
-    /// Frame was created through [`attach()`].
+    Context(&'f dyn Context),
+    /// Frame was created through [`attach()`] or [`attach_printable()`].
     ///
     /// [`attach()`]: crate::Report::attach
-    Attachment,
+    /// [`attach_printable()`]: crate::Report::attach_printable
+    Attachment(Attachment<'f>),
 }
+
+/// Classification of an attachment, determined by how it was created.
+///
+/// [`Frame`]: crate::Frame
+#[non_exhaustive]
+pub enum Attachment<'f> {
```

Comment:
`AttachmentKind` over `Attachment`?

Reply:
It's more consistent. I've always disliked the convention of using "Kind" for things anyway in Rust, so I'll shut up, and you should do what you think makes sense :)

Reply:
I'll go with `AttachmentKind` because in the case we decide to add an `Attachment` trait alias in the future, this wouldn't be a breaking change.

Reply:
> It's more consistent. I've always disliked the convention of using "Kind" for things anyway in Rust, so I'll shut up, and you should do what you think makes sense :)

TBH I also really like `Kind`, however, I still prefer it over `Type` because `type` is a reserved name you can't use for variables.